internal/newsfeed/news_feed_manager.go

```diff
@@ -96,8 +96,10 @@ func (n *NewsFeedManager) FetchRSS() ([]*gofeed.Item, error) {
 		}
 	}
 
-	// Update fetchFrom to now
-	n.fetchFrom = time.Now()
+	if len(filteredItems) > 0 {
+		// Update fetchFrom to now since we have new items
+		n.fetchFrom = time.Now()
+	}
```

Comment:
Is it worth testing this behaviour? Not sure what the best pattern would be to inspect this `fetchFrom` value, but it could be useful to test that we're conditionally updating this value. Thoughts?